src/libraries/PoolKeyChecker.sol

```diff
@@ -0,0 +1,12 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.24;
+
+import {PoolKey} from "@uniswap/v4-core/src/types/PoolKey.sol";
+
+library PoolKeyChecker {
+    // On UniswapV4, the minimum tick spacing is 1.
+    // This means that if the tick spacing is 0, the pool key has not been set.
+    function isEmpty(PoolKey memory poolKey) internal pure returns (bool) {
```

Comment:
this library is just used for 1 single function call, and its 1 line of code.
I think I'd personally rather have no library and just do it inline in PosM like

```
        // Store the poolKey if it is not already stored.
        if (poolKeys[info.poolId()].tickSpacing == 0) {
            poolKeys[info.poolId()] = poolKey;
        }
```

Reply:
I mostly liked it as a lib cause its not obvious that tickSpacing should be checked so i was just getting it out of the main codepath but i can change if you would prefer